src/core/db/repository/shift_repository.py

```diff
@@ -21,8 +21,22 @@ class ShiftRepository(AbstractRepository):
     def __init__(self, session: AsyncSession = Depends(get_session)) -> None:
         super().__init__(session, Shift)
 
-    async def get_with_users(self, id: UUID) -> Shift:
-        statement = select(Shift).where(Shift.id == id).options(selectinload(Shift.users).selectinload(User.user_tasks))
+    async def get_with_members(self, id: UUID, member_status: Optional[Member.Status]) -> Shift:
+        """Получить смену (Shift) по её id вместе со связанными данными.
+
+        Связанные данные: Shift -> Memeber -> User, Shift -> Member -> UserTask.
+
+        Аргументы:
+            id (UUID) - id смены (shift)
+            member_status (Optional[Member.Status]) - статус участника смены.
+        """
+        filter_members_by_status = or_(member_status is None, Member.status == member_status)

```

Comment:
Мне кажется, что такой вариант не очень очевиден: filter_members_by_status = or_(member_status is None, Member.status == member_status)

Предлагаю так:
```suggestion
        member_stmt = Shift.members
        if member_status:
            member_stmt = member_stmt.where(Member.status == member_status)

        statement = (
            select(Shift)
            .where(Shift.id == id)
            .options(selectinload(member_stmt).selectinload(Member.user_tasks))
            .options(selectinload(member_stmt).selectinload(Member.user))
        )
```